src/preprocess_data.py

```diff
@@ -54,18 +77,34 @@ def preprocess_data():
     characters_data['hair'] = (characters_data['hair'].str.split(' ').str[0]).astype("category")
     characters_data['sex'] = (characters_data['sex'].str.split(' ').str[0]).astype("category")
     characters_data['gsm'] = (characters_data['gsm'].str.split(' ').str[0]).astype("category")
-    characters_data['first_appearance' ]= pd.to_datetime(characters_data['first_appearance'],
-                                                        format='%b-%y',
-                                                        errors='coerce')
+
     characters_data['year'] = pd.to_datetime(characters_data['year'],
-                                            format="%Y",
-                                            errors='coerce')
+                                             format="%Y",
+                                             errors='coerce').dt.year
     characters_data['publisher'] = characters_data['publisher'].astype("category")
 
+    if verbose: print(f"Writing full data output file: {output_filename}")
     characters_data.to_csv(output_filename)
     if verbose: print("\nOutput data summary:")
     if verbose: print(f"{characters_data.info()}")
 
+    # Creating deployment data file from rows missing target values")
+    deploy_df = characters_data[characters_data['align'].isnull()]
+    deploy_df.to_csv(output_file_slug + "_deploy." + output_file_ext)
+    characters_data = characters_data[characters_data['align'].notna()]
+
+    # Split train and test data
+    train_df, test_df = train_test_split(characters_data, test_size=0.3, random_state=123)
```

Comment:
For the test_size=0.3, this is a bit higher, we may choose test_size=0.2 or 0.25?

Reply:
Also, in align we have just 3 rows with value as "Reformed", are we deleting those as well ?

Reply:
I will add a line to change reformed to "Good" in our cleaned data, and lower the test size to 0.2.